### PR TITLE
[SCB-2043] flow control support leak bucket and token bucket

### PR DESCRIPTION
**Pull request by GuoYL123**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: handlers/handler-router/src/main/java/org/apache/servicecomb/router/custom/RouterInvokeFilter.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ private boolean loadHeaders() {
   private boolean isHaveHeadersRule() {
     DynamicStringProperty headerStr = DynamicPropertyFactory.getInstance()
         .getStringProperty(SERVICECOMB_ROUTER_HEADER, null);
-    return !StringUtils.isEmpty(headerStr.get());
+    return !StringUtils.isNotEmpty(headerStr.get());
```
Comment (liubao68) [inline handlers/handler-router/src/main/java/org/apache/servicecomb/router/custom/RouterInvokeFilter.java:112]: wrong

Comment (GuoYL123) [reply handlers/handler-router/src/main/java/org/apache/servicecomb/router/custom/RouterInvokeFilter.java]: done

File: handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/FixedWindowStrategy.java [hunk 2]
```diff
@@ -14,15 +14,14 @@
...
+package org.apache.servicecomb.qps.strategy;
 
 import java.util.concurrent.atomic.AtomicLong;
 
-public class QpsController {
-  private String key;
-
-  private Integer qpsLimit;
+/**
+ * @Date 2020/7/8
```
Comment (liubao68) [inline handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/FixedWindowStrategy.java:22]: `@Date` seems no use, can be removed.

File: handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/SlidingWindowStrategy.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+package org.apache.servicecomb.qps.strategy;
+
+
+/**
+ * @Date 2020/7/8
+ **/
+public class SlidingWindowStrategy extends AbstractQpsStrategy {
+
+  public SlidingWindowStrategy(long qpsLimit, String key) {
```
Comment (liubao68) [inline handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/SlidingWindowStrategy.java:26]: Seems no used.

File: handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/FixedWindowStrategy.java (hunk 2 above)
Comment (GuoYL123) [reply handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/FixedWindowStrategy.java]: done

File: handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/SlidingWindowStrategy.java (hunk 3 above)
Comment (GuoYL123) [reply handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/strategy/SlidingWindowStrategy.java]: deleted

**Pull request merged**